scripts/xtensa-build-zephyr.py

```diff
@@ -809,6 +809,13 @@ def build_platforms():
 		if args.debug:
 			overlays.append(str(pathlib.Path(SOF_TOP, "app", "debug_overlay.conf")))
 
+		# The xt-cland Cadence toolchain currently cannot link shared
+		# libraries for Xtensa. Therefore when it's used we switch to
+		# building relocatable ELF objects.
+		if ("ZEPHYR_TOOLCHAIN_VARIANT" in platf_build_environ.keys() and
+		    platf_build_environ["ZEPHYR_TOOLCHAIN_VARIANT"] == 'xt-clang'):
```

Comment:
Python can already do that for you:
```suggestion
if platf_build_environ.get("ZEPHYR_TOOLCHAIN_VARIANT") == 'xt-clang'
```